### ISSUE TITLE
Problem with vector attribute after calling Retokenizer.split

### ISSUE THREAD

**Issue by klti** [explosion/spaCy#3540]
When trying to use split and retokenize like in the docs (https://spacy.io/usage/linguistic-features/#retokenization), `displacy.render(list(doc.sents))` breaks with an error

## How to reproduce the behaviour

code to reproduce (nearly the same as example in docs, only made strings unicode and added displacy.render calls)
```
import spacy
from spacy import displacy

nlp = spacy.load("en_core_web_sm")
doc = nlp(u"I live in NewYork right now")
print("Before:", [token.text for token in doc])
displacy.render(doc)  # works
displacy.render(doc, style="dep")  # works
displacy.render(list(doc.sents), style="dep")  # works

with doc.retokenize() as retokenizer:
    heads = [(doc[3], 1), doc[2]]
    attrs = {"POS": ["PROPN", "PROPN"], "DEP": ["pobj", "compound"]}
    retokenizer.split(doc[3], [u"New", u"York"], heads=heads, attrs=attrs)
print("After:", [token.text for token in doc])
displacy.render(doc)  # works
displacy.render(doc, style="dep")  # works
displacy.render(list(doc.sents), style="dep")  # breaks
```

Output
```
('Before:', [u'I', u'live', u'in', u'NewYork', u'right', u'now'])
('After:', [u'I', u'live', u'in', u'New', u'York', u'right', u'now'])
Traceback (most recent call last):
  File "test_example.py", line 18, in <module>
    displacy.render(list(doc.sents), style="dep") # breaks
  File "/usr/lib/python2.7/site-packages/spacy/displacy/__init__.py", line 46, in render
    docs = [obj if not isinstance(obj, Span) else obj.as_doc() for obj in docs]
  File "span.pyx", line 224, in spacy.tokens.span.Span.as_doc
  File "span.pyx", line 410, in spacy.tokens.span.Span.vector.__get__
  File "span.pyx", line 187, in __iter__
  File "token.pyx", line 392, in spacy.tokens.token.Token.vector.__get__
IndexError: index 6 is out of bounds for axis 0 with size 6
```

## Your Environment

* **Python version:** 2.7.16
* **Platform:** Linux-5.0.3-arch1-1-ARCH-x86_64-with-glibc2.2.5
* **spaCy version:** 2.1.3

P.S.: great library, I love it :thumbsup:

**Comment by ines** [comment]
Thanks for the report – based on the error, this looks like it might be a problem with how the `vector` attribute is handled after splitting.

**Comment by svlandeg** [comment]
@klti : thanks for the report! It was indeed an issue with the `vector` attribute as @ines pointed out. This should hopefully be fixed by PR #4097.
